libsyclinterface/source/dpctl_sycl_device_interface.cpp

```diff
@@ -849,3 +849,56 @@ DPCTLDevice_GetSubGroupSizes(__dpctl_keep const DPCTLSyclDeviceRef DRef,
     }
     return sizes;
 }
+
+__dpctl_give DPCTLDeviceVectorRef
+DPCTLDevice_GetComponentDevices(__dpctl_keep const DPCTLSyclDeviceRef DRef)
+{
+    using vecTy = std::vector<DPCTLSyclDeviceRef>;
+    vecTy *ComponentDevicesVectorPtr = nullptr;
+    if (DRef) {
+        auto D = unwrap<device>(DRef);
+        try {
+            auto componentDevices =
+                D->get_info<sycl::ext::oneapi::experimental::info::device::
+                                component_devices>();
+            ComponentDevicesVectorPtr = new vecTy();
```

Comment:
Would it make sense to reserve the number of elements in the vector?
```suggestion
            ComponentDevicesVectorPtr = new vecTy(componentDevices.size());
```

Reply:
Used a call to reserve afterward, to avoid default-initialization of `DPCTLSyclDeviceRef`